# Simple CMS Open Data Plot

Demonstrates a simple plot of Muon Track pT from the [CMS OpenData Higgs to 4 lepton dataset](http://opendata.cern.ch/record/1507). The dataset number (picked from the URL) is 1507.

In [1]:
from func_adl_servicex import ServiceXSourceCMSRun1AOD
from hist import Hist
import mplhep as mpl

# This is CMS data...
# mpl.style.use(mpl.style.CMS)

The following fetches the data.

In [2]:
data = ServiceXSourceCMSRun1AOD("cernopendata://16") \
    .SelectMany(lambda e: e.TrackMuons("globalMuons")) \
    .Select(lambda m: m.pt()) \
    .AsAwkwardArray(['mu_pt']) \
    .value()

cernopendata://16:   0%|          | 0/9000000000.0 [00:00]

        cernopendata://16 Downloaded:   0%|          | 0/9000000000.0 [00:00]

Transform 8d6a3691-1f5e-4bbb-9c4f-5116c6dad5d2 had 1697 errors:
  Error transforming file: root://eospublic.cern.ch//eos/opendata/cms/Run2011A/DoubleElectron/AOD/12Oct2013-v1/20000/02F78600-B13E-E311-9844-003048FF86CA.root
  -> error: Failed to transform input file root://eospublic.cern.ch//eos/opendata/cms/Run2011A/DoubleElectron/AOD/12Oct2013-v1/20000/229F2ACE-543E-E311-AFC3-003048678A78.root: Output file /home/atlas/root:::eospublic.cern.ch::eos:opendata:cms:Run2011A:DoubleElectron:AOD:12Oct2013-v1:20000:229F2ACE-543E-E311-AFC3-003048678A78.root was not found -- errors: Setting up CMSSW_5_3_32
  -> CMSSW should now be available.
  -> 
  Error transforming file: root://eospublic.cern.ch//eos/opendata/cms/Run2011A/DoubleElectron/AOD/12Oct2013-v1/20000/0224FE84-BF3E-E311-8592-002618943854.root
  -> error: Failed to transform input file root://eospublic.cern.ch//eos/opendata/cms/Run2011A/DoubleElectron/AOD/12Oct2013-v1/20000/229F2ACE-543E-E311-AFC3-003048678A78.root: Output file /home/a

ServiceXException: (ServiceXException(...), 'Failed to transform all files in 8d6a3691-1f5e-4bbb-9c4f-5116c6dad5d2')

In [ ]:
h = (Hist.new
     .Reg(50, 0, 200, name='mu_pt', label='Muon Track $p_T$')
     .Int64()
     )
h.fill(data['mu_pt'])
_ = h.plot()

Some quick notes on running:

| Dataset | Files | Time | Dataset Size | Column Size | Notes |
| :- | -: | -: | -: | -: | -: |
| [cernopendata://1507](http://opendata.cern.ch/record/1507) | 22 | 4:00 | 58.2 GB | 3 MB | |
| [cernopendata://16](http://opendata.cern.ch/record/16) | 1697 | 29:07 | 6.7 TB | 33 MB | Took 35 secs to resolve DID |
